### PR TITLE
Invalidate all cached preprocessors on changes

### PR DESCRIPTION
**Pull request by pushred**
This stops short of installed modules but invalidates everything else to ensure that preprocessor code is reloaded on the next page load. I could try to filter the modules to just those that require the changed module (children) but not sure how to do that comparison and the necessary traversal/comparisons would likely be more expensive than just invalidating everything.

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -322,9 +322,15 @@ var SolidusServer = function( options ){
 
   // updates the source of a preprocessor
   this.updatePreprocessor = function( preprocessor_path ){
+    var preprocessor_regex = new RegExp( escapePathForRegex( paths.preprocessors ) +'.+\.js', 'i' );
 
-    delete require.cache[require.resolve(preprocessor_path)];
+    var preprocessor_paths = Object.keys(require.cache).filter(function( path ){
+      return preprocessor_regex.test(path);
+    });
```
Comment (joanniclaborde) [inline lib/server.js]: You could replace all that's above with `var preprocessor_paths = _.values(preprocessors).map(function (preprocessor) {return preprocessor.path});`

File: test/solidus.js [hunk 2]
```diff
@@ -1058,6 +1058,37 @@ describe( 'Solidus', function(){
...
+                .end( function( err, res ){
+                  if( err ) throw err;
+                  assert( res.body.test === 'okok' );
+                  done();
+                });
+            }, FILESYSTEM_DELAY );
+          });
+      }, FILESYSTEM_DELAY );
+    });
+
```
Comment (joanniclaborde) [inline test/solidus.js]: I guess you can remove the `Updates preprocessors when their files change` test now?

File: lib/server.js (hunk 1 above)
Comment (pushred) [inline lib/server.js]: Oh! I missed that there was a var with them all in this scope, despite the method immediately above adding to it

File: test/solidus.js (hunk 2 above)
Comment (pushred) [inline test/solidus.js]: Yeah I wasn't sure about that, it's testing the same method though and the result will be the same

**Pull request merged**